test/integration/targets/win_psmodule/tasks/main.yml

```diff
@@ -87,6 +87,7 @@
       - "repo_fail|failed"
       - "module_check.stdout_lines[0] == '{{ powershell_module }}'"
       - "allow_clobber_check.stdout_lines[0] == '{{ allow_clobber_module }}'"
+  when: version.stdout_lines[0]|int >= 5
```

Comment:
And let me now come back to the example I referred to earlier: https://github.com/ansible/ansible/blob/devel/test/integration/targets/win_scheduled_task/tasks/main.yml

We add the tests in a separate *taskbook*, only run it when some condition is met (capability or powershell version) and then run the same tests twice, once in normal mode, and once in check-mode.

For some tasks we do not have to make a distinction, but for some tasks (idempotency tests) we do have to have different asserts.